lib/cocoapods-core/source.rb

```diff
@@ -24,6 +24,7 @@ class Source
     #
     def initialize(repo)
       @repo = Pathname(repo).expand_path
+      @name_to_versions_cache = {}
```

Comment:
`versions_by_name`?
Also, can you see any place where this would need to be invalidated?